packages/scaffold/src/partials/w3m-email-login-widget/index.ts

```diff
@@ -0,0 +1,116 @@
+import { ConnectorController } from '@web3modal/core'
+import { customElement } from '@web3modal/ui'
+import { LitElement, html } from 'lit'
+import { state } from 'lit/decorators.js'
+import { ref, createRef } from 'lit/directives/ref.js'
+import type { Ref } from 'lit/directives/ref.js'
+import styles from './styles.js'
+import { SnackController, RouterController } from '@web3modal/core'
+
+@customElement('w3m-email-login-widget')
+export class W3mEmailLoginWidget extends LitElement {
+  public static override styles = styles
+
+  // -- Members ------------------------------------------- //
+  private unsubscribe: (() => void)[] = []
+
+  private formRef: Ref<HTMLFormElement> = createRef()
+
+  // -- State & Properties -------------------------------- //
+  @state() private connectors = ConnectorController.state.connectors
+
+  @state() private email = ''
+
+  @state() private loading = false
+
+  public constructor() {
+    super()
+    this.unsubscribe.push(
+      ConnectorController.subscribeKey('connectors', val => (this.connectors = val))
+    )
+  }
+
+  public override disconnectedCallback() {
+    this.unsubscribe.forEach(unsubscribe => unsubscribe())
+  }
+
+  public override firstUpdated() {
+    this.formRef.value?.addEventListener('keydown', event => {
+      if (event.key === 'Enter') {
+        this.onSubmitEmail(event)
+      }
+    })
+  }
+
+  // -- Render -------------------------------------------- //
+  public override render() {
+    const multipleConnectors = this.connectors.length > 1
+    const connector = this.connectors.find(c => c.type === 'EMAIL')
+    const isSubmit = !this.loading && this.email.length > 3
```

Comment:
What about having an email validator instead of a length check here? It can be achieved by a regex like the following. Would having this decrease the error handling retention? Would it make sense or we could pass this mission to Magic? @0xAsimetriq

Reply:
All inputs and outputs (including email) are validated in `W3mFrameSchema.ts` file. Length check here is to determine when certain ui element is shown or hidden. You can try entering invalid email, you should get an error toast (raw unformatted messages yet however)